x-pack/test/api_integration/apis/ml/saved_objects/can_delete_trained_model.ts

```diff
@@ -0,0 +1,136 @@
+  const ml = getService('ml');
+  const spacesService = getService('spaces');
+  const supertest = getService('supertestWithoutAuth');
+
+  const modelIdSpace1 = 'dfa1_classification_model_n_0';
+  const modelIdStarSpace = 'dfa2_classification_model_n_0';
+  const idSpace1 = 'space1';
+  const idSpace2 = 'space2';
+
+  async function runRequest(
+    jobType: TrainedModelType,
+    ids: string[],
+    user: USER,
+    expectedStatusCode: number,
+    space?: string
+  ) {
+    const { body } = await supertest
+      .post(
+        `${
+          space ? `/s/${space}` : ''
+        }/api/ml/saved_objects/can_delete_ml_space_aware_item/${jobType}`
+      )
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send({ ids })
+      .expect(expectedStatusCode);
+    return body;
+  }
+
+  function getTestModel(id: string, modelType: ModelType) {
+    return {
+      model_id: id,
+      body: {
+        compressed_definition: ml.api.getCompressedModelDefinition(modelType),
+        inference_config: {
+          [modelType]: {},
+        },
+        input: {
+          field_names: ['common_field'],
+        },
+      } as PutTrainedModelConfig,
+    };
+  }
+
+  describe('POST saved_objects/can_delete_ml_space_aware_item trained models', () => {
+    before(async () => {
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      const model1 = getTestModel(modelIdSpace1, 'classification');
+      const model2 = getTestModel(modelIdStarSpace, 'classification');
+      await ml.api.createTrainedModel(model1.model_id, model1.body, idSpace1);
+      await ml.api.createTrainedModel(model2.model_id, model2.body, idSpace1);
+
+      await ml.api.updateTrainedModelSpaces(modelIdSpace1, [idSpace2], [], idSpace1);
+      await ml.api.assertTrainedModelSpaces(modelIdSpace1, [idSpace1, idSpace2]);
+
+      // mode the model to the * space
+      await ml.api.updateTrainedModelSpaces(modelIdStarSpace, ['*'], [idSpace1], idSpace1);
+      await ml.api.assertTrainedModelSpaces(modelIdStarSpace, ['*']);
+
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.api.cleanMlIndices();
+      await ml.testResources.cleanMLSavedObjects();
+    });
+
+    it('job in individual spaces, single space user can only untag', async () => {
+      const body = await runRequest(
+        'trained-model',
+        [modelIdSpace1],
+        USER.ML_POWERUSER_SPACE1,
+        200,
+        idSpace1
+      );
+
+      expect(body).to.eql({ [modelIdSpace1]: { canDelete: false, canRemoveFromSpace: true } });
+    });
+
+    it('job in individual spaces, all spaces user can delete and untag', async () => {
```

Comment:
Should this be `model in individual space...`?

Reply:
https://github.com/elastic/kibana/pull/127438/commits/11169a3de4bcaacacc677f09ee27309368104c95